### PR TITLE
Fix #28

### LINKED ISSUE #28: Nested menus don't have proper keyboard interaction.
If you a nested elements that implement this behavior, the keyboard events will bubble and cause all the nested elements to trigger their keyboard interaction.

### PR DESCRIPTION
**Pull request by blasten**
Issue #28 
cc @notwaldorf @frankiefu

### CODE REVIEW

File: iron-menu-behavior.html [hunk 1]
```diff
@@ -208,24 +208,50 @@
...
+    },
+
+    /**
      * Handler that is called when a shift+tab keypress is detected by the menu.
      *
      * @param {CustomEvent} event A key combination event.
      */
     _onShiftTabDown: function(event) {
-      var oldTabIndex;
+      if (this._isClosestNode(event.detail.keyboardEvent)) {
```
Comment (cdata) [inline iron-menu-behavior.html]: Rather than indenting the whole method implemetation, just `return` early if the condition is not `true`.

File: iron-menu-behavior.html [hunk 2]
```diff
@@ -208,24 +208,50 @@
     },
 
     /**
+     * Returns the item in the first menu from where the event was dispatched.
+     *
+     * @param {Event}
+     */
+    _isClosestNode: function(event) {
```
Comment (cdata) [inline iron-menu-behavior.html]: This method name could probably use some work. `is` suggests you are comparing the type or identity of the object passed as the first argument. `ClosestNode` suggests you are comparing values, and otherwise looking for the closest of something more general than a "menu".

Comment (blasten) [inline iron-menu-behavior.html]: yup,  I changed the purpose of this method and forgot to rename it

File: iron-menu-behavior.html (hunk 1 above)
Comment (blasten) [inline iron-menu-behavior.html]: done.

File: iron-menu-behavior.html [hunk 3]
```diff
@@ -270,7 +302,9 @@
      * @param {CustomEvent} event A key combination event.
      */
     _onDownKey: function(event) {
-      this._focusNext();
+      if (this._getClosestItem(event.detail.keyboardEvent)) {
```
Comment (cdata) [inline iron-menu-behavior.html]: This kind of seems like an inversion of what `stopPropagation` would do for us. Perhaps we need to re-consider that approach. WDYT?

Comment (blasten) [inline iron-menu-behavior.html]: @cdata SGTM. it's now using stopPropagation.

**Comment by blasten** [comment]
@cdata  Now it's using `stopPropagation`. Please also check https://github.com/PolymerElements/paper-menu/pull/51

**Comment by cdata** [comment]
@blasten Thanks, this is looking a lot better. Since this fixes a bug, can you add a test so that we can catch regressions in the future?

**Comment by blasten** [comment]
@cdata sure will do that!  Tests added.

File: test/iron-menu-behavior.html [hunk 4]
```diff
@@ -100,6 +113,53 @@
           }, 200);
         });
 
+        test('keyboard events should not bubble', function(done) {
+          var menu = fixture('nested');
+          var keyCounter = 0;
+          var firstNestedMenu = menu.firstElementChild;
+
+          function triggerKeyEvent(keyCode, shift) {
```
Comment (cdata) [inline test/iron-menu-behavior.html]: Can we use MockInteractions for this instead? Or if it's not sufficient, can we update it to include the functionality we need?

Comment (blasten) [inline test/iron-menu-behavior.html]: oh. it seems like this event now bubbles in MockInteractions: https://github.com/PolymerElements/iron-test-helpers/blob/master/mock-interactions.js#L147. 

 I updated the test. Also, I found that a few of the old tests are failing in Firefox. I can't reproduce the issue locally, but it seems like a timing issue.  This issue is present in #master as well.

File: test/iron-menu-behavior.html [hunk 5]
```diff
@@ -21,7 +21,7 @@
 
     <script src="../../web-component-tester/browser.js"></script>
     <script src="../../test-fixture/test-fixture-mocha.js"></script>
```
Comment (cdata) [inline test/iron-menu-behavior.html]: This is orthogonal to the change, but can you set the `web-component-tester` version to `v3.4.0` and remove `test-fixture-mocha.js` and `test-fixture.html` from the test suite?

**Comment by cdata** [comment]
@blasten can you rebase this branch?

File: test/iron-menu-behavior.html (hunk 5 above)
Comment (blasten) [inline test/iron-menu-behavior.html]: sounds good

**Comment by cdata** [comment]
@blasten bump

**Comment by blasten** [comment]
@cdata will do :)

**Comment by blasten** [comment]
@cdata rebase done!

**Comment by cdata** [comment]
LGTM :+1:
